libs/dh/message-archive/feature-log-search/src/lib/dh-message-archive-log-search.component.ts

```diff
@@ -232,7 +235,6 @@ export class DhMessageArchiveLogSearchComponent {
 
   resetSearchCritera() {
     this.store.resetState();
-    //TODO: reset form not working
-    this.searchForm.reset();
+    this.searchForm = buildSearchForm();
```

Comment:
issue: interesting solution. This doesn't really reset the `FormGroup` assigned to `searchForm` initially but rather creates a new `FormGroup` instance with initial values. It works but it feels more like a workaround.

I took a look at the documentation for [FormControl](https://angular.io/api/forms/FormControl#formcontrol) and found something interesting. When creating a new `FormControl`, an extra second param object can be passed that takes a few values. One of them is ` nonNullable`.

By looking at it, it seems like adding it will make the form control reset to its initial value on reset and not `null`, as we saw a few days ago.

![image](https://user-images.githubusercontent.com/1096332/208894097-53cb0e40-0c1f-4afd-8f4f-bb745b3348a1.png)

They also have [an example in the docs](https://angular.io/api/forms/FormControl#reset-the-control-to-its-initial-value):

![image](https://user-images.githubusercontent.com/1096332/208894240-94e7fa9d-64ef-4994-9dbe-0a75ee2f7880.png)

They also have an example for how to reset a FormControl to specific value:

![image](https://user-images.githubusercontent.com/1096332/208894738-33efde93-bb34-4da9-88fa-8e7c18036f04.png)

So maybe we don't need the adjustments to our component. What we need to do instead is make sure we set `{nonNullable: true}` on those FormControls that have an initial value different than `null`.

Reply:
![image](https://user-images.githubusercontent.com/1250254/212324618-2c6fc8da-d7e8-4371-99be-bcc3b96d5cc2.png)

Array gets set to null even though it is set to nonNullable: true

![image](https://user-images.githubusercontent.com/1250254/212324745-f8d0eaa6-6b65-4ee6-91e9-0616153594bb.png)